src/main/java/Student.java

```diff
@@ -0,0 +1,146 @@
+
+import java.time.DayOfWeek;
+import java.time.Duration;
+import java.time.LocalDateTime;
+
+	public class Student{
+		private String studentName;
+		private Curriculum curriculum;
+		private LocalDateTime startDateTime;
+
+		
+		public Student(String studentName, Curriculum curriculum, LocalDateTime startDateTime) {
+			this.studentName = studentName;
+			this.curriculum = curriculum;
+			this.startDateTime = startDateTime;
+		}
+
+		
+	public String getStudentName() {
+		return studentName;
+	}
+
+	
+	public Curriculum getCurriculum() {
+		return curriculum;
+	}
+
+	
+	public LocalDateTime getStartDateTime() {
+		return startDateTime;
+	}
+	
+	
+	public LocalDateTime getEndDateTime(LocalDateTime startDateTime, Curriculum curriculum) {
+		int remainingHours = curriculum.getTotalDurationHrs(); 
+        final int hoursPerDay = 8; 
+        LocalDateTime endDateTime = startDateTime;
+
+        while (remainingHours > 0) {
+            // Check if it's a work day 
+            if (endDateTime.getDayOfWeek() != DayOfWeek.SATURDAY && endDateTime.getDayOfWeek() != DayOfWeek.SUNDAY) {
+                // Check if it's within working hours
+                if (endDateTime.getHour() >= 10 && endDateTime.getHour() < 18) {
+                    int hoursToAdd = Math.min(hoursPerDay, remainingHours); // Add 8 hours or remaining hours, whichever is smaller
+                    endDateTime = endDateTime.plusHours(hoursToAdd);
+                    remainingHours -= hoursToAdd;
+                } else {
+                    // Move to the next day at 10 am 
+                	endDateTime = endDateTime.plusDays(1).withHour(10).withMinute(0);
+                }
+            } else {
+                // Move to the next day at 10 am
+            	endDateTime = endDateTime.plusDays(1).withHour(10).withMinute(0);
+            }
+        }
+
+        return endDateTime;
+    }
+	
+    
+        public void getProgress(LocalDateTime startDateTime, Curriculum curriculum) {
+        	LocalDateTime currentDateTime = LocalDateTime.now();
+        	LocalDateTime endDateTime = getEndDateTime(startDateTime, curriculum); 
+        	
+        	int diff1 = currentDateTime.compareTo(startDateTime);
+        	int diff2 = currentDateTime.compareTo(endDateTime); 
+        	
+        	if (diff1 >= 0) { //training has been started
+                if (diff2 < 0) { //training not completed
+                    printProgressNotFinished(startDateTime, curriculum);
+                } else { //training is completed
+                    printProgressCompleted(startDateTime, curriculum);
+                }
+            } else {
+                printProgressBeforeStart(startDateTime, curriculum);
+            }
```

Comment:
this is hard to read and hard to debug. I would strongly advice to not nest if/else structures. Also it's better to have meaningful names instead of comments. For example you can extract diff1 >= 0 to method with nice name